### ISSUE TITLE
NuGet package is not signed

### ISSUE THREAD

**Issue by NinjaCross** [mcintyre321/OneOf#119]
Hi.
Unfortunately it's not possible to use this amazing package in any signed application, since the DLLs contained into the NuGet package are not signed theirselves.
I'm aware of the existence of https://www.nuget.org/packages/OneOf.Signed/, but my stakeholders don't like the idea of using a "non official" package.
Could it be possible to sign the DLLs, so that we can use the official package of OneOf ?

**Comment by NinjaCross** [comment]
Hi :)
Any news on this ?

**Comment by AraHaan** [comment]
by signing, you mean strong name right?

**Comment by AraHaan** [comment]
oh you do, OneOf.Signed is outdated too rip...

This means this is a duplicate of this issue, https://github.com/mcintyre321/OneOf/issues/122

**Comment by AraHaan** [comment]
@mcintyre321 is it ok if I pr in an snk file so that way it will unblock me on my projects which are themselves strong named?

**Comment by mcintyre321** [comment]
I'm concerned that it will cause me maintenance issues (and thus cost me time) in the future.

Can you use https://github.com/dsplaisted/strongnamer ?

**Comment by AraHaan** [comment]
It might work, however it might not because on my application I crossgen the dependencies first by using the sharedframework .NET SDK from dotnet/arcade in order to also allow faster startup times (which also that .NET SDK requires strong naming as well).

**Comment by NinjaCross** [comment]
> I'm concerned that it will cause me maintenance issues (and thus cost me time) in the future.
> 
> Can you use https://github.com/dsplaisted/strongnamer ?

Unfortunately no, it's against my customer's policy to rely on non-signed external resources, and self-signing an external resource would be interpreted as a form of "tampering".
I know, it's a technicality, but real-world applications are often driven and regulated by very strict criteria like this.

**Comment by NinjaCross** [comment]
@AraHaan did you close this issue on purpose ?
I see that you closed also #122, so know there are no *open* issues related to this (still unresolved, AFAIK) topic

**Comment by AraHaan** [comment]
I did not close this issue, but my issue that I later discovered was a duplicate of this one.

**Comment by NinjaCross** [comment]
Sorry @AraHaan , for some reason GitHub was showing this issue as closed too.
Now I see it opened :)

**Comment by mcintyre321** [comment]
I'm afraid this isn't going to happen without some kind of commercial support being offered from the consumer.

**Comment by AraHaan** [comment]
But snk files are fully free to create, anyone can create them.

**Comment by mcintyre321** [comment]
As I understand it, publicly releasing the SNK subverts the guarantees that strong naming gives, so no client who cares about strong naming should be satisfied using a project with a public SNK. 

Managing the SNK privately in a safe consistent fashion (i.e. without accidentally losing it) is non-trivial, and opens me up to issues with clients all having to recompile their apps should the SNK get lost.

**Comment by AraHaan** [comment]
Not really, there is a such thing as having the public repo store only the public key and then store the private key somewhere else where github actions can only have access to it so that way only official builds would be able to sign with it.

**Comment by AArnott** [comment]
> As I understand it, publicly releasing the SNK subverts the guarantees that strong naming gives, so no client who cares about strong naming should be satisfied using a project with a public SNK.

This is not true. But rather than carry on this conversation on a closed issue, I'll post over at #192
